.github/workflows/mintlify.yml

```diff
@@ -0,0 +1,46 @@
+name: Migrate to Mintlify
+
+on:
+  workflow_call:
+    
+  push:
+    branches:
+      - main
+    paths:
+      - migration/**.mjs
+      - server/**.ts
+
+jobs:
+  run-migration:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate GitHub token
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.MINTLIFY_MIGRATION_APP_ID }}
+          private-key: ${{ secrets.MINTLIFY_MIGRATION_PRIVATE_KEY }}
+
+      - name: Check out mintlify branch
+        uses: actions/checkout@v4
+        with:
+          ref: mintlify
+
+      - name: Run the migration script
+        run: |
+          yarn
+          yarn git-update
+          yarn build-node
+          yarn mintlify
```

Comment:
We're currently doing all this on the mintlify branch, as it was checked out immediately above. I think we want to pull in changes from `main` first, then rebuild.